### PR TITLE
Series options should be updated as well as data

### PR DESCRIPTION
**Pull request by RuslanZavacky**
Currently, there was a limitation in ember-highcharts, that prevents passing series options to the Highchart element. 
This limitation, prevents from passing important options like `pointStart` and `pointInterval` as example.

Here is our content:

```javascript
content: computed('data', function() {
    return [{
      name:  'Name of the first chart',
      color: HighchartsTheme.colors[0],
      data: current,
      pointStart: point,
      pointInterval: pointInterval,
      type: 'area'
    }, {
      name: 'Name of the second chart',
      color: HighchartsTheme.colors[1],
      data: previous,
      pointStart: point,
      pointInterval: pointInterval,
      type: 'line',
      zoneAxis: 'x',
      zones: zones
    }];
  })
```

So, if any of the options, apart from `data` changes, this changes where not reflected in the chart, because chart.redraw do not take options into account, unless you specify so.

----

In this PR we are checking if there are more options, than 1, assuming, that data is always present, and if there is more, then we call `series.update`, instead of `series.setData`. It might have some drawbacks, as Highcharts API treats .update quite "destructively":

> Update the series with a new set of options. For a clean and precise handling of new options, all methods and elements from the series is removed, and it is initiated from scratch. Therefore, this method is more performance expensive than some other utility methods like setData or setVisible.

### CODE REVIEW

File: addon/components/high-charts.js [hunk 1]
```diff
@@ -73,7 +73,11 @@ export default Component.extend({
         return chartSeriesToRemove.push(series);
       }
 
-      series.setData(contentSeries.data, false);
+      if (Object.keys(contentSeries).length > 1) {
```
Comment (gregjopa) [inline addon/components/high-charts.js]: This seems like it is going to evaluate to true the majority of use cases. Most `series` have at least two keys:
```js
{
  name: 'Jane',
  data: [5, 0, 20]
}
```

As you mentioned `series.update` is an expensive operation so I'd like to avoid it whenever possible. 

In the dummy app in this repo we have an interactive line chart example for manually testing the "update data" feature. I tested that with your change and `series.update` always gets called. 

Perhaps we can do a diff between `contentSeries` and `chart.series`. If only the `data` value changed then call `setData`. Else call `series.update`.

Comment (RuslanZavacky) [inline addon/components/high-charts.js]: Sure, I'll improve it, to validate if data changed.

**Comment by RuslanZavacky** [comment]
@gregjopa can you check it now :) With validation of previous it takes a bit more of implementation. 

Also, I've removed total property from commits data, as it was added on the initial load, but then never used anywhere. And as we compare current vs previous, previous was initially different, because of total.

**Comment by RuslanZavacky** [comment]
oh, looks like our use-case now stopped working.. need to investigate more

**Comment by RuslanZavacky** [comment]
Fixed, forgot to save current state to previous, when re-draw

**Comment by RuslanZavacky** [comment]
@gregjopa don't want to be too persuasive, what about new solution? ) We've battle tested it in our application, seems like it works much better now.

**Comment by gregjopa** [comment]
This is a good start. But I don't think we need lodash. I also think we can solve this problem in a simpler way. Here's my understanding of the problem:

```js
// new data
var newData = {
  name:  'Name of the first chart',
  data: [1, 2, 3, 4, 5],
  pointStart: Date.UTC(2016, 1, 1),
  pointInterval: 24 * 3600 * 1000,
  type: 'area'
};


// series from existing highcharts object
var chartSeries =  {
  name:  'Name of the first chart',
  data: [1, 2, 3, 4, 5],
  pointStart: Date.UTC(2016, 4, 1),
  pointInterval: 24 * 3600 * 1000,
  type: 'area',
  _sysProp1: 123,
  _sysProp2: 456
};


// Problem: determine which key values in "newData" are different from "chartSeries".
// The "data" key can be ignored and only primitive data types are supported (strings, numbers, and dates)
``` 

The _sysProp keys don't matter in `chartSeries` since we are only doing a one-way diff. Also, the main series attributes are primitive values (numbers, strings, and booleans): http://api.highcharts.com/highcharts#series. I recommend keeping it simple for now and only supporting primitive data types in this diffing process. Lodash diffing is great for doing deep compares but for our use case we don't need it since we are sticking to primitive checking.

So based on these restricted requirements I came up with this solution:

```js
var updatedKeys = Object.keys(contentSeries).filter(key => {
  var isDataKey = key === 'data';
  var isValidType = ['object', 'function'].indexOf(typeof contentSeries[key]) === -1;
  var isTheSame = contentSeries[key] === series[key];

  return !isDataKey && isValidType && !isTheSame;
});

// call series.update() when other series attributes like pointStart have changed
if (updatedKeys.length) {
  series.update(contentSeries, false);
}
else {
  series.setData(contentSeries.data, false);
}
```

code: https://github.com/gregjopa/ember-highcharts/commit/fbd2474527e3daf937e8d082f5c0922bfa793e45

Please let me know if this code works for your use case.

**Comment by RuslanZavacky** [comment]
@gregjopa one issue that I see here, is that we now assume, that we work only with primitives, which is not alway true. When you want to build complex tooltips, you will pass complex data into Highcharts via dataset array of objects. And, when we do this type of a comparison, it might be hard for people to track errors, if we presumably ignore edge cases.

I will test your approach, but I bet that performance and size of the plugin won't be super big depending on the use of lodash or not :)

And about system fields, they matter on the initial load, as initial object will be "clean", without any _* props, and the new one, will have them. So ignoring this fields will automatically trigger full reload, when it might not be needed.

**Comment by gregjopa** [comment]
That's a good point about the _* properties. I missed that. I added a commit to handle it: https://github.com/gregjopa/ember-highcharts/commit/31bda22d673c31c1b54733675ded68dc2dce1d99

This addon is already missing a lot of test coverage so I fear adding a lot more code and functionality without tests. My thought is to keep this PR minimal to support the main use cases.

**Comment by RuslanZavacky** [comment]
@gregjopa looks like your implementation working :) we can proceed with your version and close this PR.

**Comment by gregjopa** [comment]
@RuslanZavacky thanks for your help. I opened PR #92 for this work.
